docs: ✏️ Add how to debug binary to docs (#1628)
hashicorp/boundary-ui: ui/desktop/README.md

More information on [debugging the main process](https://www.electronjs.org/docs/latest/tutorial/debugging-main-process).
More information on [Node debugging guide](https://nodejs.org/en/docs/guides/debugging-getting-started/).

### Debug `binary`

> The desktop client binary is the end artifact we deliver to customers. More information on how to build such binary in the [Building for Production](#building-for-production).

Reviewer: ```suggestion
The desktop client binary is the end artifact we deliver to customers. More information on how to build the binary in [Building for Production](#building-for-production).
```

> We use [Debugtron](https://github.com/pd4d10/debugtron) for debug the production binary. Debugtron let you inspect the Electron app as if you were on a dev environment. Also provides debug options for the node process running Electron.

Reviewer: ```suggestion
We use [Debugtron](https://github.com/pd4d10/debugtron) to debug the production binary. Debugtron let's you inspect the Electron app as if you were in a dev environment. It also provides debug options for the node process running Electron.
```
